src/System.Reflection.MetadataLoadContext/src/System/Reflection/MetadataLoadContext.Apis.cs

```diff
@@ -0,0 +1,302 @@
+    //
+    // Once an assembly has been bound, no assembly with the same assembly name identity
+    // can be bound again from a different location unless the MVID's are identical.
+    //
+    // Once loaded, the underlying file may be locked for the duration of the MetadataLoadContext's lifetime. You can
+    // release the locks by disposing the MetadataLoadContext object. The behavior of any Type, Assembly or other reflection
+    // objects handed out by the MetadataLoadContext is undefined after disposal. Though objects provided by the MetadataLoadContext
+    // strive to throw an ObjectDisposedException, this is not guaranteed. Some apis may return fixed or previously
+    // cached data. Accessing objects *during* a Dispose may result in a unmanaged access violation and failfast.
+    //
+    // Comparing Type, Member and Assembly objects:
+    //
+    //   The right way to compare two Reflection objects dispensed by the MetadataLoadContext are:
+    //
+    //       m1 == m2
+    //       m1.Equals(m2)
+    //
+    //   but not
+    //
+    //       object.ReferenceEquals(m1, m2)   // WRONG
+    //       (object)m1 == (object)m2         // WRONG
+    //
+    //   Note that the following descriptions are not literal descriptions of how Equals() is implemented. The MetadataLoadContext
+    //   reserves the right to implement Equals() as "object.ReferenceEquals()" and intern the associated objects in such
+    //   a way that Equals() works "as if" it were comparing those things.
+    //
+    // - Each MetadataLoadContext permits only one Assembly instance per assembly identity so equality of assemblies is the same as the
+    //   equality of their assembly identity.
+    // 
+    // - Modules are compared by comparing their containing assemblies and their row indices in the assembly's manifest file table.
+    // 
+    // - Defined types are compared by comparing their containing modules and their row indices in the module's TypeDefinition table.
+    // 
+    // - Constructed types (arrays, byrefs, pointers, generic instances) are compared by comparing all of their component types.
+    // 
+    // - Generic parameter types are compared by comparing their containing Modules and their row indices in the module's GenericParameter table.
+    // 
+    // - Constructors, methods, fields, events and properties are compared by comparing their declaring types, their row indices in their respective
+    //   token tables and their ReflectedType property.
+    // 
+    // - Parameters are compared by comparing their declaring member and their position index.
+    //
+    // Multithreading:
+    //    
+    //   The MetadataLoadContext and the reflection objects it hands out are all multithread-safe and logically immutable,
+    //   with the following provisos:
+    //
+    //   - Adding (or removing) handlers to the MetadataLoadContext's events is not thread-safe. These should be done prior
```

Comment:
This comment is no longer applicable. The events are no longer in the public surface.

Reply:
Thanks. I'll do another pass on this to ensure I didn't miss any more